### PR TITLE
Update Smoke Tests for October GA release

### PR DESCRIPTION
**Pull request by danieljurek**
_Do not merge until https://github.com/Azure/azure-sdk-for-net/pull/8135 is merged and this branch is rebased properly_

Update smoke tests to work with GA packages

### CODE REVIEW

**Comment by danieljurek** [comment]
@Azure/azure-sdk-eng

File: common/SmokeTests/SmokeTest/SmokeTest.csproj [hunk 1]
```diff
@@ -6,12 +6,12 @@
...
-    <PackageReference Include="Azure.Messaging.EventHubs" Version="5.0.0-preview.4" />
-    <PackageReference Include="Azure.Security.Keyvault.Secrets" Version="4.0.0-preview.5" />
-    <PackageReference Include="Azure.Storage.Blobs" Version="12.0.0-preview.4" />
-    <PackageReference Include="Microsoft.Azure.DocumentDB.Core" Version="2.8.1" />
-    <PackageReference Include="System.Reflection.Metadata" Version="1.6.0" />
+    <PackageReference Include="Azure.Identity" Version="1.0.0" />
+    <PackageReference Include="Azure.Messaging.EventHubs" Version="5.0.0-preview.5" />
+    <PackageReference Include="Azure.Security.Keyvault.Secrets" Version="4.0.0" />
+    <PackageReference Include="Azure.Storage.Blobs" Version="12.0.0" />
+    <PackageReference Include="Microsoft.Azure.DocumentDB.Core" Version="2.9.0" />
```
Comment (weshaggard) [inline common/SmokeTests/SmokeTest/SmokeTest.csproj]: Is there a reason we need to specify these 2 dependencies explicitly?

**Review by weshaggard — APPROVED** [review]

File: common/SmokeTests/SmokeTest/SmokeTest.csproj [hunk 2]
```diff
@@ -6,12 +6,12 @@
...
-    <PackageReference Include="Azure.Messaging.EventHubs" Version="5.0.0-preview.4" />
-    <PackageReference Include="Azure.Security.Keyvault.Secrets" Version="4.0.0-preview.5" />
-    <PackageReference Include="Azure.Storage.Blobs" Version="12.0.0-preview.4" />
-    <PackageReference Include="Microsoft.Azure.DocumentDB.Core" Version="2.8.1" />
-    <PackageReference Include="System.Reflection.Metadata" Version="1.6.0" />
+    <PackageReference Include="Azure.Identity" Version="1.0.0" />
+    <PackageReference Include="Azure.Messaging.EventHubs" Version="5.0.0-preview.5" />
+    <PackageReference Include="Azure.Security.Keyvault.Secrets" Version="4.0.0" />
+    <PackageReference Include="Azure.Storage.Blobs" Version="12.0.0" />
+    <PackageReference Include="Microsoft.Azure.DocumentDB.Core" />
```
Comment (weshaggard) [inline common/SmokeTests/SmokeTest/SmokeTest.csproj]: Still curious about these last to references but it isn't worth blocking the PR. Why do we need them explicitly and why do we not have a version for them?

Comment (danieljurek) [reply common/SmokeTests/SmokeTest/SmokeTest.csproj]: DocumentDB.Core -- We wanted to have bones of a CosmosDB smoke test in place and used the existing libraries to get there. 

System.Reflection.Metadata -- I remember there being some error that was mitigated by using this library but a recent [run with the smoke tests removed](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=204908&view=logs&j=ebc9d770-2e6f-5ce9-65da-8c05f5cedaa7&t=887f72bf-6086-5d1c-045d-069210869763) shows that it's no longer needed... Pr to fix it: https://github.com/Azure/azure-sdk-for-net/pull/9021

**Pull request merged**